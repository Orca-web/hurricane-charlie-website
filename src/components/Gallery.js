import React, { Component } from 'react';
import GalleryItem from './GalleryItem';
import CallToAction from './CallToAction';
import Modal from './Modal';

class Gallery extends Component {
  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    const { galleryData, modalStatus, modalHandler } = this.props;

    const gallery = galleryData 
      ? (<section className="gallery">
            <div className="gallery-grid">
              {galleryData.map(item => (
                <GalleryItem modalHandler={modalHandler} galleryItemData={item} key={item.id} />
              ))}
            </div>
          </section>) 
      : null;

    const galleryContent = modalStatus.open 
      ? <Modal 
          modalData={galleryData[modalStatus.id]} 
          modalHandler={modalHandler}
          modalLimit={galleryData.length - 1}
        /> 
      : gallery;

    return (
      <div>
        {galleryContent}
        <CallToAction />
      </div>
    )
  }
}

export default Gallery