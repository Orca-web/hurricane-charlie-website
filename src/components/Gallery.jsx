import React, { Component } from 'react'

class Gallery extends Component {
  render() {
    return (
      <section className="gallery">
        <div className="gallery-grid">
          <div className="grid-item large"></div>
          <div className="grid-item wide"></div>
          <div className="grid-item wide"></div>
          <div className="grid-item tall"></div>
          <div className="grid-item tall"></div>
          <div className="grid-item small"></div>
          <div className="grid-item small"></div>
          <div className="grid-item large"></div>
          <div className="grid-item wide"></div>
        </div>
      </section>
    )
  }
}

export default Gallery