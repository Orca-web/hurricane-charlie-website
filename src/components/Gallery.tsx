import React, { Component } from 'react';
import GalleryItem from './GalleryItem';
import CallToAction from './CallToAction';
import Modal from './Modal';
import { galleryData } from '../globals/galleryData';

import imageImport from '../helpers/imageImport';

interface GalleryState {
    selectedGalleryItemId: number;
    isModalShowing: boolean;
}

const images = imageImport(require.context('../images/thumbs', false, /\.jpg$/)) as { [key: string]: string };

class Gallery extends Component<{}, GalleryState> {
    state = {
        selectedGalleryItemId: 0,
        isModalShowing: false,
    }

    componentDidMount() {
        window.scrollTo(0, 0)
    }

    selectGalleryItem = (id: number) => {
        this.setState({ selectedGalleryItemId: id, isModalShowing: true })
    }

    modalHandler = (command: string): void => {
        let { selectedGalleryItemId } = this.state;
        switch (command) {
            case "next":
                this.setState({ selectedGalleryItemId: selectedGalleryItemId + 1 });
                break;
            case "prev":
                this.setState({ selectedGalleryItemId: selectedGalleryItemId - 1 });
                break;
            default:
                this.setState({ isModalShowing: false });
                break;
        }
    }

    render() {
        const { isModalShowing, selectedGalleryItemId } = this.state;

        if (isModalShowing) {
            return (
                <Modal
                    selectedGalleryItemId={selectedGalleryItemId}
                    modalHandler={this.modalHandler}
                    modalLimit={galleryData.length - 1}
                /> 
            )
        }

        return (
            <div>
                <section className="gallery">
                    <div className="gallery-grid">
                        {galleryData.map(item => (
                            <GalleryItem
                                selectGalleryItem={this.selectGalleryItem}
                                galleryItemData={item}
                                key={item.id}
                                imgData={images[item.src]}
                            />
                        ))}
                    </div>
                </section>
                <CallToAction />
            </div>
        )
    }
}

export default Gallery