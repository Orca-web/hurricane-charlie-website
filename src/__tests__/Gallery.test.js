import React from "react";
import { shallow, configure } from "enzyme";
import Gallery from "../components/Gallery";

const app = shallow(<Gallery />);

describe('Gallery', () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});
