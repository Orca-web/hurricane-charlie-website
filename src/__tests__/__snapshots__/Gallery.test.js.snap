// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gallery />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="gallery"
        >
          <div
            className="gallery-grid"
          />
        </section>,
        <CallToAction />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="gallery-grid"
          />,
          "className": "gallery",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "gallery-grid",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="gallery"
          >
            <div
              className="gallery-grid"
            />
          </section>,
          <CallToAction />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="gallery-grid"
            />,
            "className": "gallery",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "gallery-grid",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
